crypto/src/symkey/mod.rs

```diff
@@ -59,6 +59,11 @@ impl SymmetricKey {
         Self { key }
     }
 
+    pub fn new_chacha20_from_bytes(bytes: [u8; KEY_LEN]) -> Self {
+        let key = SymmetricKeyHolder::XChacha20Poly1305(Chacha20poly1305Key::new_from_array(bytes));
+        Self { key }
+    }
```

Comment:
You don't need this, unless I'm missing something. You have `Decode` implemented/derived for this struct. So, something like `SymmetricKey::decode_all(&bytes)` should work.

Notice you use `decode_all` to consume all the bytes given, while you sue `decode` to consume only what's needed to create the current object.

Reply:
maybe the name is confusing, it is not meant for decoding from bytes, it is creating new key using those bytes as key, the bytes being the ones obtained from the password.

Reason was the other new method does not take bytes, it generates random ones.
The KEY_LEN was put as that is what the Chacha20poly1305Key::new_from_array() was taking, otherwise an error would need to be returned.

Reply:
Thinking about this for a minute, I don't see a way around having an error. The problem is that this interface is generic and isn't only for chacha.

So, I recommend creating a method here: `from_raw_key(KeyKind, bytes_slice)`, and depending on the key kind, we enforce the slice size.

Reply:
if it is a runtime decision for which `KeyKind` to use yes, but if it is known at compile time which one to use wouldn't it be ok to have a `from_raw_key_specific_algo(bytes_slice)` so you avoid the error/using except("")?

Reply:
It's not that big of a deal to have some error here... interface consistency is more important in this case, I'd say.

Reply:
changed to `from_raw_key(KeyKind, bytes_slice)` that returns an error for invalid key bytes length